app/controllers/items_controller.rb

```diff
@@ -0,0 +1,60 @@
+class ItemsController < ApplicationController
+  before_action :set_item, only: [:show, :edit, :update, :destroy]
+  def index
+    if params[:merchant_id]
+      @merchant = Merchant.find(params[:merchant_id])
+      @items = @merchant.items
+    else
+      @items = Item.all
+    end
+    @items = sort_items(@items, params[:sort]) if params[:sort]
```

Comment:
We could argue on whether or not you call this sort_items in your view or not, granted you make this sort_items a model behavior instead of a method in the controller. I'm actually okay with you calling it in the controller, but like I said, the method below should be something that you have in your model. It would likely be a class method since you're organizing a lot of items, so I'd have the class method take in a possible merchant so that you can sort by merchant if need be within that class method.